app/notification.go

```diff
@@ -613,6 +613,16 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 					}
 					message.Add("thread", string(payload))
 
+					previousUnreadMentions := userThread.UnreadMentions
+					previousUnreadReplies := userThread.UnreadReplies - 1
```

Comment:
At first I thought this would always be >= 1, since we are receiving a new message, 
but that's not the case of poster sending the message AKA `userThread.UnreadReplies == 0`, 
the client handles it, but still the server should never send negative values.